client/components/date-range/trigger.tsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent } from 'react';
+import { noop } from 'lodash';
+import Gridicon from 'components/gridicon';
+import { useTranslate } from 'i18n-calypso';
+import { Moment } from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import ButtonGroup from 'components/button-group';
+import ScreenReaderText from 'components/screen-reader-text';
+import SharedProps from './shared-props';
+
+interface Props {
+	startDate: Date | Moment | null | undefined;
+	endDate: Date | Moment | null | undefined;
+	startDateText: string;
+	endDateText: string;
+	buttonRef: object;
+	onTriggerClick: () => void;
+	onClearClick: () => void;
+	triggerText: ( startDateText: string, endDateText: string ) => string;
+	showClearBtn: boolean;
+}
+
+const DateRangeTrigger: FunctionComponent< Props & SharedProps > = props => {
+	const translate = useTranslate();
+
+	const { startDate, endDate, showClearBtn, startDateText, endDateText, triggerText } = props;
+
+	const canReset = startDate || endDate;
```

Comment:
This looks like a boolean, but it's not 🙂 `true` and `false` aren't even possible values 🙃 (type is currently inferred as `Date | Moment | null | undefined`)

If we annotate, that becomes apparent:
```ts
	const canReset: boolean = startDate || endDate; // Type error!
```

Shall we ensure it is?

```suggestion
	const canReset = Boolean( startDate || endDate );
```